consent-ws/src/main/java/org/broadinstitute/consent/http/service/EmailNotifierService.java

```diff
@@ -478,4 +499,22 @@ private String retrieveElectionTypeStringCollect(String electionType) {
         }
         return ElectionTypeString.DATA_ACCESS.getValue();
     }
+
+    private List<String> getAcademicEmails(List<DACUser> users) {
+        List<String> academicEmails = new ArrayList<>();
+        for(DACUser user: users) {
+            if (dacUserDAO.checkResearcherRole(user.getDacUserId()) != null) {
+                List<ResearcherProperty> researcherProperties = researcherPropertyDAO.findResearcherPropertiesByUser(user.getDacUserId());
+                Optional<ResearcherProperty> checkNotification = researcherProperties.stream().filter(rp -> rp.getPropertyKey().equals(ResearcherFields.CHECK_NOTIFICATIONS.getValue())).findFirst();
+                if(checkNotification.isPresent()){
+                    if (checkNotification.get().getPropertyValue().equals("true")) {
+                        ResearcherProperty academicEmailRP = researcherProperties.stream().filter(rp -> rp.getPropertyKey().equals(ResearcherFields.ACADEMIC_BUSINESS_EMAIL.getValue())).findFirst().get();
+                        academicEmails.add(academicEmailRP.getPropertyValue());
+                    }
+                }
+
+            }
+        }
+        return academicEmails;
```

Comment:
The structure here makes me nervous. Inside a loop, we make a db call to check the role, then we make another db call to found the researcher properties. Is the `checkResearcherRole` call necessary if we're following this up with a query on researcher properties? Doesn't that second query mean the properties are for a researcher? It would make more sense to me to: 

- Batch query for all researcher properties for user ids in the DACUser list
- Inner join that on the role table to enforce researcher role
- Filter the query on property key in (CHECK_NOTIFICATIONS, ACADEMIC_BUSINESS_EMAIL)
- Iterate over the users
  - For properties that match check notifications and have a non-empty email, add to list.

Reply:
I completely agree with you, I changed it